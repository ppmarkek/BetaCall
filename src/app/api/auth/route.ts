import axios from 'axios';

const authURL = 'https://localhost:8080/api/users';

export const userSignIn = async ({
  email,
  password,
}: {
  email: string;
  password: string;
}) => {
  try {
    const response = await axios.post(`${authURL}/login`, { email, password });
    return response.data;
  } catch (err) {
    return console.error(err);
  }
};
