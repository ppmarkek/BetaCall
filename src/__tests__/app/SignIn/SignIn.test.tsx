jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(() => new URLSearchParams('')),
  useRouter: jest.fn(() => ({ push: jest.fn() })),
}));

jest.mock('@/app/api/auth/route', () => ({
  userSignIn: jest.fn(),
  userAppwriteSignIn: jest.fn(),
}));

jest.mock('@/lib/appwrite', () => ({
  account: {
    createOAuth2Session: jest.fn(),
    get: jest.fn(),
  },
}));

import React from 'react';
import { fireEvent, render, screen, waitFor } from '../../../../test-utils';
import '@testing-library/jest-dom';
import SignInPage from '@/app/signIn/page';
import { AxiosHeaders, AxiosResponse } from 'axios';
import { userSignIn, userAppwriteSignIn } from '@/app/api/auth/route';
import { account } from '@/lib/appwrite';
import { OAuthProvider } from 'appwrite';
import { useRouter, useSearchParams } from 'next/navigation';

const emptyHeaders = new AxiosHeaders();

describe('Sign In Page', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    (useSearchParams as jest.Mock).mockReturnValue(new URLSearchParams(''));
  });

  it('matches the snapshot', () => {
    const { container } = render(<SignInPage />);
    expect(container).toMatchSnapshot();
  });

  it('shows spinner on form submit and hides it after request resolves', async () => {
    let resolveRequest!: (value: AxiosResponse<{ message: string }>) => void;
    const requestPromise = new Promise<AxiosResponse<{ message: string }>>(
      (resolve) => {
        resolveRequest = resolve;
      }
    );
    (userSignIn as jest.Mock).mockReturnValue(requestPromise);

    render(<SignInPage />);

    const emailInput = screen.getByPlaceholderText('Enter your email');
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });

    const passwordInput = screen.getByPlaceholderText('Enter your password');
    fireEvent.change(passwordInput, { target: { value: '{S[6O$}V56v^' } });

    const submitButton = screen.getByTestId('button-submit');
    fireEvent.click(submitButton);

    await waitFor(() => {
      expect(screen.getByTestId('spinner')).toBeInTheDocument();
    });

    resolveRequest({
      data: { message: 'Authentication successful' },
      status: 200,
      statusText: 'OK',
      headers: emptyHeaders,
      config: { headers: emptyHeaders },
    });

    await waitFor(() => {
      expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
    });
  });

  it('sets error message when bad request with status 403', async () => {
    let resolveRequest!: (value: AxiosResponse<{ message: string }>) => void;
    const requestPromise = new Promise<AxiosResponse<{ message: string }>>(
      (resolve) => {
        resolveRequest = resolve;
      }
    );
    (userSignIn as jest.Mock).mockReturnValue(requestPromise);

    render(<SignInPage />);

    const emailInput = screen.getByPlaceholderText('Enter your email');
    fireEvent.change(emailInput, { target: { value: 'testFail@example.com' } });

    const passwordInput = screen.getByPlaceholderText('Enter your password');
    fireEvent.change(passwordInput, { target: { value: '{S[6O$}V56v^' } });

    const submitButton = screen.getByTestId('button-submit');
    fireEvent.click(submitButton);

    resolveRequest({
      data: { message: 'Invalid or expired password reset token' },
      status: 403,
      statusText: 'Forbidden',
      headers: emptyHeaders,
      config: { headers: emptyHeaders },
    });
  });

  it('sets error message when bad request with status 401', async () => {
    let resolveRequest!: (value: AxiosResponse<{ message: string }>) => void;
    const requestPromise = new Promise<AxiosResponse<{ message: string }>>(
      (resolve) => {
        resolveRequest = resolve;
      }
    );
    (userSignIn as jest.Mock).mockReturnValue(requestPromise);

    render(<SignInPage />);

    const emailInput = screen.getByPlaceholderText('Enter your email');
    fireEvent.change(emailInput, { target: { value: 'testFail@example.com' } });

    const passwordInput = screen.getByPlaceholderText('Enter your password');
    fireEvent.change(passwordInput, { target: { value: '{S[6O$}V56v^' } });

    const submitButton = screen.getByTestId('button-submit');
    fireEvent.click(submitButton);

    resolveRequest({
      data: { message: 'Invalid credentials' },
      status: 401,
      statusText: 'Unauthorized',
      headers: emptyHeaders,
      config: { headers: emptyHeaders },
    });

    await waitFor(() => {
      expect(
        screen.getByText('Incorrect email or password')
      ).toBeInTheDocument();
    });
  });

  describe('OAuth2 session creation', () => {
    it('calls createOAuth2Session for Google login', () => {
      (useSearchParams as jest.Mock).mockReturnValue(new URLSearchParams(''));
      render(<SignInPage />);
      const googleButton = screen.getByText('Sign In with Google');
      fireEvent.click(googleButton);
      expect(account.createOAuth2Session).toHaveBeenCalledWith(
        OAuthProvider.Google,
        'http://localhost:3000/signIn?socialMedia=true',
        'http://localhost:3000/signIn?socialMedia=false'
      );
    });

    it('calls createOAuth2Session for GitHub login', () => {
      (useSearchParams as jest.Mock).mockReturnValue(new URLSearchParams(''));
      render(<SignInPage />);
      const githubButton = screen.getByText('Sign Up with GitHub');
      fireEvent.click(githubButton);
      expect(account.createOAuth2Session).toHaveBeenCalledWith(
        OAuthProvider.Github,
        'http://localhost:3000/signIn?socialMedia=true',
        'http://localhost:3000/signIn?socialMedia=false'
      );
    });
  });

  describe('Social media sign-in useEffect', () => {
    it('handles social sign-in with status 200', async () => {
      const pushMock = jest.fn();
      (useRouter as jest.Mock).mockReturnValue({ push: pushMock });
      const searchParams = new URLSearchParams('socialMedia=true&other=1');
      (useSearchParams as jest.Mock).mockReturnValue(searchParams);

      const userData = {
        email: 'user@example.com',
        $id: '123',
        name: 'John Doe',
      };
      (account.get as jest.Mock).mockResolvedValue(userData);
      (userAppwriteSignIn as jest.Mock).mockResolvedValue({
        status: 200,
        data: { accessToken: 'access', refreshToken: 'refresh' },
      });
      const replaceStateSpy = jest
        .spyOn(window.history, 'replaceState')
        .mockImplementation(() => {});

      render(<SignInPage />);
      await waitFor(() => {
        expect(pushMock).toHaveBeenCalledWith('/');
      });
      expect(replaceStateSpy).toHaveBeenCalledWith(null, '', '?other=1');
      replaceStateSpy.mockRestore();
    });

    it('handles social sign-in with status 403', async () => {
      const pushMock = jest.fn();
      (useRouter as jest.Mock).mockReturnValue({ push: pushMock });
      const searchParams = new URLSearchParams('socialMedia=true');
      (useSearchParams as jest.Mock).mockReturnValue(searchParams);

      const userData = {
        email: 'user@example.com',
        $id: '123',
        name: 'John Doe',
      };
      (account.get as jest.Mock).mockResolvedValue(userData);
      (userAppwriteSignIn as jest.Mock).mockResolvedValue({
        status: 403,
        data: {},
      });

      render(<SignInPage />);
      await waitFor(() => {
        expect(pushMock).toHaveBeenCalledWith(`/verify/${userData.email}`);
      });
    });

    it('handles social sign-in with status 404', async () => {
      const pushMock = jest.fn();
      (useRouter as jest.Mock).mockReturnValue({ push: pushMock });
      const searchParams = new URLSearchParams('socialMedia=true');
      (useSearchParams as jest.Mock).mockReturnValue(searchParams);

      const userData = {
        email: 'user@example.com',
        $id: '123',
        name: 'John Doe',
      };
      (account.get as jest.Mock).mockResolvedValue(userData);
      (userAppwriteSignIn as jest.Mock).mockResolvedValue({
        status: 404,
        data: {},
      });

      render(<SignInPage />);
      await waitFor(() => {
        expect(pushMock).toHaveBeenCalledWith(
          `/signUp?socialMedia=true&step=2&email=${encodeURIComponent(
            userData.email
          )}&appwriteId=${encodeURIComponent(userData.$id)}&firstName=${encodeURIComponent(
            'John'
          )}&lastName=${encodeURIComponent('Doe')}`
        );
      });
    });

    it('check new password visibility', () => {
      render(<SignInPage />);
      const passwordInput = screen.getByPlaceholderText(
        'Enter your password'
      ) as HTMLInputElement;

      expect(passwordInput.type).toBe('password');

      const icons = screen.getAllByTestId('icon-textInput');
      const newPasswordToggleIcon = icons[1];
      fireEvent.click(newPasswordToggleIcon);

      expect(passwordInput.type).toBe('text');
    });

    it('logs error when account.get() fails with an error not including "missing scope"', async () => {
      const consoleErrorSpy = jest
        .spyOn(console, 'error')
        .mockImplementation(() => {});
      const pushMock = jest.fn();
      (useRouter as jest.Mock).mockReturnValue({ push: pushMock });
      const searchParams = new URLSearchParams('socialMedia=true');
      (useSearchParams as jest.Mock).mockReturnValue(searchParams);

      const error = new Error('Test error');
      (account.get as jest.Mock).mockRejectedValue(error);

      render(<SignInPage />);
      await waitFor(() => {
        expect(consoleErrorSpy).toHaveBeenCalledWith(
          'Error fetching google account data:',
          error
        );
      });
      consoleErrorSpy.mockRestore();
    });
  });
});
