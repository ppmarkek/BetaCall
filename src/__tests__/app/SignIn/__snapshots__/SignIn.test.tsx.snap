// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Page matches the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc(100svh - 81px);
  min-height: 800px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 420px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  gap: 50px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
}

.emotion-3 {
  font-family: Lato;
  font-weight: 700;
  font-size: 32px;
  line-height: 42px;
}

.emotion-4 {
  color: #8083A3;
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 35px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
}

.emotion-6 .input-icon {
  fill: #8083a3;
  -webkit-transition: fill 0.3s;
  transition: fill 0.3s;
}

.emotion-6:focus-within .input-icon:not(.input-icon-error) {
  fill: #1a1c1d;
}

.emotion-6 .input-icon-error {
  fill: #ff808b;
}

.emotion-7 {
  color: #8083A3;
  font-family: Lato;
  font-size: 16px;
  font-weight: 400;
  line-height: 19px;
}

.emotion-8 {
  color: #1a1c1d;
  padding: 20px 0px;
  padding-left: 0px;
  padding-right: 25px;
  border: unset;
  border-radius: unset;
  border-bottom: 1px solid #eceef5;
  color: #1a1c1d;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

@layer recipes {
  .emotion-8 {
    width: 100%;
    min-width: var(--input-height);
    outline: 0;
    position: relative;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    text-align: start;
    border-radius: var(--chakra-radii-l2);
    height: var(--input-height);
    --focus-color: var(--chakra-colors-color-palette-focus-ring);
    --error-color: var(--chakra-colors-border-error);
    font-size: var(--chakra-font-sizes-sm);
    line-height: 1.25rem;
    padding-inline: var(--chakra-spacing-3);
    --input-height: var(--chakra-sizes-10);
    background: var(--chakra-colors-transparent);
    --bg-currentcolor: var(--chakra-colors-transparent);
    border-width: 1px;
    border-color: var(--chakra-colors-border);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
  }

  .emotion-8:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-8:is([data-invalid], [aria-invalid=true], [data-state=invalid]) {
    --focus-ring-color: var(--error-color);
    border-color: var(--error-color);
  }

  .emotion-8:is(:focus-visible, [data-focus-visible]) {
    outline-offset: 0px;
    outline-width: var(--focus-ring-width, 1px);
    outline-color: var(--focus-ring-color);
    outline-style: var(--focus-ring-style, solid);
    border-color: var(--focus-ring-color);
  }
}

.emotion-8:focus-visible {
  outline: unset;
  border-bottom: 1px solid #000;
}

.emotion-9 {
  position: absolute;
  right: 0;
  bottom: 8px;
  font-size: 20px;
  cursor: default;
}

@layer recipes {
  .emotion-9 {
    display: inline-block;
    line-height: 1em;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    color: currentcolor;
    vertical-align: middle;
  }
}

.emotion-13 {
  position: absolute;
  right: 0;
  bottom: 8px;
  font-size: 20px;
  cursor: pointer;
}

@layer recipes {
  .emotion-13 {
    display: inline-block;
    line-height: 1em;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    color: currentcolor;
    vertical-align: middle;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

.emotion-16 {
  -webkit-appearance: none;
  background: #f0f0f3;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  cursor: pointer;
  position: relative;
  outline: none;
}

.emotion-16:checked {
  background: #6b59cc;
}

.emotion-16:checked::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background: #fff;
}

.emotion-17 {
  font-family: Lato;
  font-size: 14px;
  font-weight: 400;
  line-height: 21px;
}

.emotion-18 {
  font-family: 'Lato',sans-serif;
  font-size: 16px;
  font-weight: 700;
  line-height: 19px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6b59cc;
  cursor: pointer;
}

.emotion-19 {
  background-color: #6B59CC;
  --bg-currentcolor: #6B59CC;
  height: 50px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  font-family: 'Lato',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  padding: 0 30px;
  border-radius: 10px;
  width: 100%;
  border: 1px solid #6B59CC;
  background: #6B59CC;
  color: #fff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

@layer recipes {
  .emotion-19 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    position: relative;
    border-radius: var(--chakra-radii-l2);
    white-space: nowrap;
    vertical-align: middle;
    border-width: 1px;
    border-color: var(--chakra-colors-transparent);
    cursor: var(--chakra-cursor-button);
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    outline: 0;
    line-height: 1.25rem;
    isolation: isolate;
    font-weight: var(--chakra-font-weights-medium);
    transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,translate,transform;
    transition-duration: var(--chakra-durations-moderate);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
    height: var(--chakra-sizes-10);
    min-width: var(--chakra-sizes-10);
    font-size: var(--chakra-font-sizes-sm);
    padding-inline: var(--chakra-spacing-4);
    gap: var(--chakra-spacing-2);
    background: var(--chakra-colors-color-palette-solid);
    --bg-currentcolor: var(--chakra-colors-color-palette-solid);
    color: var(--chakra-colors-color-palette-contrast);
  }

  .emotion-19:is(:focus-visible, [data-focus-visible]) {
    outline-width: var(--focus-ring-width, 2px);
    outline-offset: var(--focus-ring-offset, 2px);
    outline-style: var(--focus-ring-style, solid);
    outline-color: var(--focus-ring-color);
  }

  .emotion-19:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-19 :where(svg) {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    width: var(--chakra-sizes-5);
    height: var(--chakra-sizes-5);
  }

  .emotion-19:is([aria-expanded=true], [data-expanded], [data-state=expanded]) {
    --mix-background: color-mix(in srgb, var(--chakra-colors-color-palette-solid) 90%, transparent);
    background: var(--mix-background, var(--chakra-colors-color-palette-solid));
    --bg-currentcolor: var(--mix-background, var(--chakra-colors-color-palette-solid));
  }

  @media (hover: hover) {
    .emotion-19:is(:hover, [data-hover]):not(:disabled, [data-disabled]) {
      --mix-background: color-mix(in srgb, var(--chakra-colors-color-palette-solid) 90%, transparent);
      background: var(--mix-background, var(--chakra-colors-color-palette-solid));
      --bg-currentcolor: var(--mix-background, var(--chakra-colors-color-palette-solid));
    }
  }
}

.emotion-19:hover {
  background: unset;
  color: #6B59CC;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-20::before,
.emotion-20::after {
  content: '';
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 1px;
  background-color: #eeeeee;
}

.emotion-20::before {
  margin-right: 12px;
  max-width: 100px;
}

.emotion-20::after {
  margin-left: 12px;
  max-width: 100px;
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  gap: 10px;
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #eceef5;
  height: 50px;
  width: 50px;
  border-radius: 10px;
  border-top-right-radius: unset;
  border-bottom-right-radius: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-25 {
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  font-family: 'Lato',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding: 0 15px;
  border-radius: 10px;
  border-top-left-radius: unset;
  border-bottom-left-radius: unset;
  width: calc(100% - 50px);
  height: 50px;
  border: 1px solid #eceef5;
  border-left: unset;
  background: unset;
  color: #8083a3;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  cursor: pointer;
}

@layer recipes {
  .emotion-25 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-appearance: none;
    -moz-appearance: none;
    -ms-appearance: none;
    appearance: none;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    position: relative;
    border-radius: var(--chakra-radii-l2);
    white-space: nowrap;
    vertical-align: middle;
    border-width: 1px;
    border-color: var(--chakra-colors-transparent);
    cursor: var(--chakra-cursor-button);
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    outline: 0;
    line-height: 1.25rem;
    isolation: isolate;
    font-weight: var(--chakra-font-weights-medium);
    transition-property: background-color,border-color,color,fill,stroke,opacity,box-shadow,translate,transform;
    transition-duration: var(--chakra-durations-moderate);
    --focus-ring-color: var(--chakra-colors-color-palette-focus-ring);
    height: var(--chakra-sizes-10);
    min-width: var(--chakra-sizes-10);
    font-size: var(--chakra-font-sizes-sm);
    padding-inline: var(--chakra-spacing-4);
    gap: var(--chakra-spacing-2);
    background: var(--chakra-colors-color-palette-solid);
    --bg-currentcolor: var(--chakra-colors-color-palette-solid);
    color: var(--chakra-colors-color-palette-contrast);
  }

  .emotion-25:is(:focus-visible, [data-focus-visible]) {
    outline-width: var(--focus-ring-width, 2px);
    outline-offset: var(--focus-ring-offset, 2px);
    outline-style: var(--focus-ring-style, solid);
    outline-color: var(--focus-ring-color);
  }

  .emotion-25:is(:disabled, [disabled], [data-disabled], [aria-disabled=true]) {
    opacity: 0.5;
    cursor: not-allowed;
  }

  .emotion-25 :where(svg) {
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    width: var(--chakra-sizes-5);
    height: var(--chakra-sizes-5);
  }

  .emotion-25:is([aria-expanded=true], [data-expanded], [data-state=expanded]) {
    --mix-background: color-mix(in srgb, var(--chakra-colors-color-palette-solid) 90%, transparent);
    background: var(--mix-background, var(--chakra-colors-color-palette-solid));
    --bg-currentcolor: var(--mix-background, var(--chakra-colors-color-palette-solid));
  }

  @media (hover: hover) {
    .emotion-25:is(:hover, [data-hover]):not(:disabled, [data-disabled]) {
      --mix-background: color-mix(in srgb, var(--chakra-colors-color-palette-solid) 90%, transparent);
      background: var(--mix-background, var(--chakra-colors-color-palette-solid));
      --bg-currentcolor: var(--mix-background, var(--chakra-colors-color-palette-solid));
    }
  }
}

.emotion-25:hover {
  background: #8083a3;
  color: #eceef5;
  border-color: #8083a3;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-3"
        >
          Tell us more about yourself
        </p>
        <p
          class="emotion-4"
        >
          Enter your details to proceed further
        </p>
      </div>
      <form>
        <div
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            <p
              class="emotion-7"
            >
              Email
            </p>
            <input
              class="chakra-input chakra-tooltip__trigger emotion-8"
              data-part="trigger"
              data-scope="tooltip"
              data-state="closed"
              dir="ltr"
              id="tooltip::r0::trigger"
              name="email"
              placeholder="Enter your email"
              type="email"
            />
            <svg
              aria-hidden="true"
              class="chakra-icon input-icon emotion-9"
              data-testid="icon-textInput"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
          </div>
          <div
            class="emotion-6"
          >
            <p
              class="emotion-7"
            >
              Password
            </p>
            <input
              class="chakra-input chakra-tooltip__trigger emotion-8"
              data-part="trigger"
              data-scope="tooltip"
              data-state="closed"
              dir="ltr"
              id="tooltip::r1::trigger"
              name="password"
              placeholder="Enter your password"
              type="password"
            />
            <svg
              aria-hidden="true"
              class="chakra-icon input-icon emotion-13"
              data-testid="icon-textInput"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
              />
            </svg>
          </div>
          <div
            class="emotion-14"
          >
            <div
              class="emotion-15"
            >
              <input
                class="emotion-16"
                name="rememberMe"
                type="checkbox"
              />
              <p
                class="emotion-17"
              >
                Remember me
              </p>
            </div>
            <a
              class="emotion-18"
              href="/recoverPassword"
            >
              Recover password
            </a>
          </div>
          <button
            class="chakra-button emotion-19"
            data-testid="button-submit"
            type="submit"
          >
            Sign In
          </button>
        </div>
      </form>
      <div
        class="emotion-20"
      >
        <p
          class="emotion-4"
        >
          Or
        </p>
      </div>
      <div
        class="emotion-22"
      >
        <div
          class="emotion-23"
        >
          <div
            class="emotion-24"
          >
            <svg
              data-testid="icon-component"
              fill="#8083A3"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 488 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
              />
            </svg>
          </div>
          <button
            class="chakra-button emotion-25"
            type="button"
          >
            Sign In with Google
          </button>
        </div>
        <div
          class="emotion-23"
        >
          <div
            class="emotion-24"
          >
            <svg
              data-testid="icon-component"
              fill="#8083A3"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </div>
          <button
            class="chakra-button emotion-25"
            type="button"
          >
            Sign Up with GitHub
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
